### PR TITLE
Improve the API documentation and markdown files

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
$subject
Related to: https://github.com/ballerina-platform/ballerina-standard-library/issues/584

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: crypto-ballerina/Package.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 ## Package Overview
 
-This package provides the necessary utilities that are required to hash content using different hashing mechanisms and algorithms. 
+The Crypto library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-For information on the operations, which you can perform with this package, see the below **Functions**. For an example on the usage of the operations, see the [Cryptographic Operations Example](https://ballerina.io/learn/by-example/crypto.html).
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data etc. with different cryptographic algorithms.
+
+For more information go to [The Crypto Module](https://docs.central.ballerina.io/ballerina/crypto/latest).
```
Comment (daneshk) [inline crypto-ballerina/Package.md:7]: Shall we remove the API doc link from Package.MD

Comment (ldclakmal) [reply crypto-ballerina/Package.md]: Fixed with https://github.com/ballerina-platform/module-ballerina-crypto/pull/172/commits/0c61412e2f212771b1e88a6271356c7d89ff2d03

File: README.md [hunk 2]
```diff
@@ -88,6 +88,6 @@ All contributors are encouraged to read the [Ballerina Code of Conduct](https://
 
 ## Useful Links
 
-* Discuss about code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
+* Discuss the code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```
Comment (MadhukaHarith92) [inline README.md:91]:
```suggestion
* Discuss code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```

File: crypto-ballerina/Module.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+## Overview
+
+This module provides the common cryptographic mechanisms with different algorithms.
+
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data etc. with different cryptographic algorithms.
+
+### Hashes
+
+Crypto module supports to generate hashes with 5 different hash algorithms MD5, SHA1, SHA256, SHA384 and SHA512. Also, it supports to generate the CRC32B checksum.
```
Comment (MadhukaHarith92) [inline crypto-ballerina/Module.md:9]:
```suggestion
Crypto module supports generating hashes with 5 different hash algorithms MD5, SHA1, SHA256, SHA384, and SHA512. Also, it supports generating the CRC32B checksum.
```

File: crypto-ballerina/Module.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+### HMAC
+
+Crypto module supports to generate HMAC with 5 different hash algorithms MD5, SHA1, SHA256, SHA384 and SHA512.
```
Comment (MadhukaHarith92) [inline crypto-ballerina/Module.md:13]:
```suggestion
Crypto module supports generating HMAC with 5 different hash algorithms MD5, SHA1, SHA256, SHA384, and SHA512.
```

File: crypto-ballerina/Module.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+### Decode Private/Public Key
+
+Crypto module supports to decode RSA private key from a `.p12` file and a key file in `PEM` format. Also, supports to decode a public key from a `.p12` file and a certificate file in `X509` format. Additionally, this supports to build an RSA public key with the modulus and the exponent parameters.
```
Comment (MadhukaHarith92) [inline crypto-ballerina/Module.md:17]:
```suggestion
Crypto module supports decoding RSA private key from a `.p12` file and a key file in `PEM` format. Also, supports decoding a public key from a `.p12` file and a certificate file in `X509` format. Additionally, this supports building an RSA public key with the modulus and the exponent parameters.
```

File: crypto-ballerina/Module.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+### Encrypt and Decrypt
+
+Crypto module supports to both symmetric key encryption/decryption and asymmetric key encryption/decryption. RSA algorithm can be used for symmetric key encryption/decryption with the use of the private and public keys. AES algorithm can be used for asymmetric key encryption/decryption with the use of shared key.
```
Comment (MadhukaHarith92) [inline crypto-ballerina/Module.md:21]:
```suggestion
Crypto module supports both symmetric key encryption/decryption and asymmetric key encryption/decryption. RSA algorithm can be used for symmetric key encryption/decryption with the use of private and public keys. AES algorithm can be used for asymmetric key encryption/decryption with the use of shared key.
```

File: crypto-ballerina/Package.md [hunk 7]
```diff
@@ -1,5 +1,15 @@
 ## Package Overview
 
-This package provides the necessary utilities that are required to hash content using different hashing mechanisms and algorithms. 
+The Crypto library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-For information on the operations, which you can perform with this package, see the below **Functions**. For an example on the usage of the operations, see the [Cryptographic Operations Example](https://ballerina.io/learn/by-example/crypto.html).
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data etc. with different cryptographic algorithms.
```
Comment (MadhukaHarith92) [inline crypto-ballerina/Package.md:5]:
```suggestion
The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data, etc. with different cryptographic algorithms.
```

File: crypto-ballerina/Module.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+Crypto module supports decoding RSA private key from a `.p12` file and a key file in `PEM` format. Also, supports decoding a public key from a `.p12` file and a certificate file in `X509` format. Additionally, this supports building an RSA public key with the modulus and the exponent parameters.
+
+### Encrypt and Decrypt
+
+Crypto module supports both symmetric key encryption/decryption and asymmetric key encryption/decryption. RSA algorithm can be used for symmetric key encryption/decryption with the use of private and public keys. AES algorithm can be used for asymmetric key encryption/decryption with the use of shared key.
+
+### Sign and Verify
+
+Crypto module supports for signing data using the RSA private key and verification of signature using the RSA public key. This supports MD5, SHA1, SHA256, SHA384 and SHA512 digesting algorithms as well.
```
Comment (anupama-pathirage) [inline crypto-ballerina/Module.md:25]:
```suggestion
Crypto module supports for signing data using the RSA private key and verification of signature using the RSA public key. This supports MD5, SHA1, SHA256, SHA384, and SHA512 digesting algorithms as well.
```

File: crypto-ballerina/Module.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Crypto module supports generating HMAC with 5 different hash algorithms MD5, SHA1, SHA256, SHA384, and SHA512.
+
+### Decode Private/Public Key
+
+Crypto module supports decoding RSA private key from a `.p12` file and a key file in `PEM` format. Also, supports decoding a public key from a `.p12` file and a certificate file in `X509` format. Additionally, this supports building an RSA public key with the modulus and the exponent parameters.
+
+### Encrypt and Decrypt
+
+Crypto module supports both symmetric key encryption/decryption and asymmetric key encryption/decryption. RSA algorithm can be used for symmetric key encryption/decryption with the use of private and public keys. AES algorithm can be used for asymmetric key encryption/decryption with the use of shared key.
```
Comment (anupama-pathirage) [inline crypto-ballerina/Module.md:21]:
```suggestion
Crypto module supports both symmetric key encryption/decryption and asymmetric key encryption/decryption. RSA algorithm can be used for symmetric key encryption/decryption with the use of private and public keys. AES algorithm can be used for asymmetric key encryption/decryption with the use of a shared key.
```

File: crypto-ballerina/Module.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
+## Overview
+
+This module provides the common cryptographic mechanisms with different algorithms.
+
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data etc. with different cryptographic algorithms.
```
Comment (anupama-pathirage) [inline crypto-ballerina/Module.md:5]:
```suggestion
The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data, etc. with different cryptographic algorithms.
```

File: README.md [hunk 11]
```diff
@@ -3,21 +3,21 @@ Ballerina Crypto Library
...
   [![codecov](https://codecov.io/gh/ballerina-platform/module-ballerina-crypto/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerina-crypto)
 
-The Crypto library is one of the standard library modules of the<a target="_blank" href="https://ballerina.io/"> Ballerina</a> language.
+The Crypto library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-It provides the necessary utilities that are required to hash content using different hashing mechanisms and algorithms.
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data, etc. with different cryptographic algorithms.
 
-For more information go to [The Encoding Module](https://ballerina.io/learn/api-docs/ballerina/crypto/index.html).
+For more information go to [The Crypto Module](https://docs.central.ballerina.io/ballerina/crypto/latest).
```
Comment (praneesha) [inline README.md:14]:
```suggestion
For more information go to the [Crypto module](https://docs.central.ballerina.io/ballerina/crypto/latest).
```

File: crypto-ballerina/Module.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data, etc. with different cryptographic algorithms.
+
+### Hashes
+
+Crypto module supports generating hashes with 5 different hash algorithms MD5, SHA1, SHA256, SHA384, and SHA512. Also, it supports generating the CRC32B checksum.
+
+### HMAC
+
+Crypto module supports generating HMAC with 5 different hash algorithms MD5, SHA1, SHA256, SHA384, and SHA512.
```
Comment (praneesha) [inline crypto-ballerina/Module.md:13]:
```suggestion
The `crypto` module supports generating HMAC with 5 different hash algorithms: MD5, SHA1, SHA256, SHA384, and SHA512.
```

File: crypto-ballerina/Module.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+### Decode Private/Public Key
+
+Crypto module supports decoding RSA private key from a `.p12` file and a key file in `PEM` format. Also, supports decoding a public key from a `.p12` file and a certificate file in `X509` format. Additionally, this supports building an RSA public key with the modulus and the exponent parameters.
```
Comment (praneesha) [inline crypto-ballerina/Module.md:17]:
```suggestion
The `crypto` module supports decoding the RSA private key from a `.p12` file and a key file in the `PEM` format. Also, it supports decoding a public key from a `.p12` file and a certificate file in the `X509` format. Additionally, this supports building an RSA public key with the modulus and exponent parameters.
```

File: crypto-ballerina/Module.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+### Encrypt and Decrypt
+
+Crypto module supports both symmetric key encryption/decryption and asymmetric key encryption/decryption. RSA algorithm can be used for symmetric key encryption/decryption with the use of private and public keys. AES algorithm can be used for asymmetric key encryption/decryption with the use of a shared key.
```
Comment (praneesha) [inline crypto-ballerina/Module.md:21]:
```suggestion
The `crypto` module supports both symmetric key encryption/decryption and asymmetric key encryption/decryption. The RSA algorithm can be used for symmetric-key encryption/decryption with the use of private and public keys. The AES algorithm can be used for asymmetric-key encryption/decryption with the use of a shared key.
```

File: crypto-ballerina/Module.md [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+### Sign and Verify
+
+Crypto module supports for signing data using the RSA private key and verification of signature using the RSA public key. This supports MD5, SHA1, SHA256, SHA384, and SHA512 digesting algorithms as well.
```
Comment (praneesha) [inline crypto-ballerina/Module.md:25]:
```suggestion
The `crypto` module supports signing data using the RSA private key and verification of the signature using the RSA public key. This supports MD5, SHA1, SHA256, SHA384, and SHA512 digesting algorithms as well.
```

File: crypto-ballerina/Package.md [hunk 16]
```diff
@@ -1,5 +1,14 @@
 ## Package Overview
 
-This package provides the necessary utilities that are required to hash content using different hashing mechanisms and algorithms. 
+The Crypto library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
```
Comment (praneesha) [inline crypto-ballerina/Package.md:3]:
```suggestion
The `crypto` library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
```

File: crypto-ballerina/Package.md [hunk 17]
```diff
@@ -1,5 +1,14 @@
...
 
-For information on the operations, which you can perform with this package, see the below **Functions**. For an example on the usage of the operations, see the [Cryptographic Operations Example](https://ballerina.io/learn/by-example/crypto.html).
+The Ballerina Crypto module facilitates APIs to do operations like hashing, HMAC generation, checksum generation, encryption, decryption, digitally signing data and verifying digitally signed data, etc. with different cryptographic algorithms.
```
Comment (praneesha) [inline crypto-ballerina/Package.md:5]:
```suggestion
The Ballerina `crypto` module facilitates APIs to perform operations like hashing, HMAC generation, checksum generation, encryption, decryption, signing data digitally, verifying digitally signed data, etc., with different cryptographic algorithms.
```

File: crypto-ballerina/hmac.bal [hunk 18]
```diff
@@ -16,86 +16,86 @@
...
+# byte[] data = input.toBytes();
+# string secret = "some-secret";
+# byte[] key = secret.toBytes();
+# byte[] hmac = crypto:hmacMd5(data, key);
 # ```
 #
 # + input - Value to be hashed
 # + key - Key used for HMAC generation
-# + return - HMAC output or `crypto:Error`
+# + return - HMAC output or `crypto:Error` if an error occurred
```
Comment (praneesha) [inline crypto-ballerina/hmac.bal:30]:
```suggestion
# + return - The HMAC output or a `crypto:Error` if an error occurred
```

File: crypto-ballerina/hmac.bal [hunk 19]
```diff
@@ -16,86 +16,86 @@
...
+# byte[] data = input.toBytes();
+# string secret = "some-secret";
+# byte[] key = secret.toBytes();
+# byte[] hmac = crypto:hmacSha1(data, key);
 # ```
 #
 # + input - Value to be hashed
 # + key - Key used for HMAC generation
-# + return - HMAC output or `crypto:Error`
+# + return - HMAC output or `crypto:Error` if an error occurred
```
Comment (praneesha) [inline crypto-ballerina/hmac.bal:47]:
```suggestion
# + return - The HMAC output or a `crypto:Error` if an error occurred
```

File: crypto-ballerina/hmac.bal [hunk 20]
```diff
@@ -16,86 +16,86 @@
...
+# byte[] data = input.toBytes();
+# string secret = "some-secret";
+# byte[] key = secret.toBytes();
+# byte[] hmac = crypto:hmacSha256(data, key);
 # ```
 #
 # + input - Value to be hashed
 # + key - Key used for HMAC generation
-# + return - HMAC output or `crypto:Error`
+# + return - HMAC output or `crypto:Error` if an error occurred
```
Comment (praneesha) [inline crypto-ballerina/hmac.bal:64]:
```suggestion
# + return - The HMAC output or a `crypto:Error` if an error occurred
```

File: crypto-ballerina/hmac.bal [hunk 21]
```diff
@@ -16,86 +16,86 @@
...
+# byte[] data = input.toBytes();
+# string secret = "some-secret";
+# byte[] key = secret.toBytes();
+# byte[] hmac = crypto:hmacSha384(data, key);
 # ```
 #
 # + input - Value to be hashed
 # + key - Key used for HMAC generation
-# + return - HMAC output or `crypto:Error`
+# + return - HMAC output or `crypto:Error` if an error occurred
```
Comment (praneesha) [inline crypto-ballerina/hmac.bal:81]:
```suggestion
# + return - The HMAC output or a `crypto:Error` if an error occurred
```

File: crypto-ballerina/hmac.bal [hunk 22]
```diff
@@ -16,86 +16,86 @@
...
+# byte[] data = input.toBytes();
+# string secret = "some-secret";
+# byte[] key = secret.toBytes();
+# byte[] hmac = crypto:hmacSha512(data, key);
 # ```
 #
 # + input - Value to be hashed
 # + key - Key used for HMAC generation
-# + return - HMAC output or `crypto:Error`
+# + return - HMAC output or `crypto:Error` if an error occurred
```
Comment (praneesha) [inline crypto-ballerina/hmac.bal:98]:
```suggestion
# + return - The HMAC output or a `crypto:Error` if an error occurred
```

**Review by MadhukaHarith92 — APPROVED** [review]
LGTM

**Review by praneesha — APPROVED** [review]

**Pull request merged**